app/assets/javascripts/sitewide/atwho_res.js.erb

```diff
@@ -0,0 +1,580 @@
+    function _templateEval(_tpl, map) {
+      var res;
+      try {
+        res = generateTemplate(map);
+      } catch (_error) {
+        res = '';
+      }
+      return res;
+    }
+
+    function _matchHighlighter(li, query, search_filter) {
+      var re, li2, prevVal, newVal;
+      if (!query) {
+        return li;
+      }
+      resourcesChecker(query, search_filter);
+      li2 = $(li);
+      re = new RegExp(query, 'gi');
+    prevVal =
+        li2
+        .find('[data-val=name]')
+        .html();
+      newVal =
+        prevVal
+        .replace(re, '<strong>$&</strong>');
+      li2.find('[data-val=name]').html(newVal);
+      newVal =
+        prevVal
+        .replace(re, '<strong>$&</strong>');
+      li2.find('[data-val=email]').html(newVal);
+      return li2[0].outerHTML;
+    }
+
+    function _genrateInputTag(value, li) {
+      var res = '';
+      res += '[#' + li.attr('data-name');
+      res += '~' + li.attr('data-type');
+      res += '~' + li.attr('data-id') + ']';
+      return res;
+    }
+
+    // check if query has some hits and disables the buttons without it
+    function resourcesChecker(query, search_filter) {
+      var src_btn, $element;
+
+      switch (search_filter) {
+        case 'task#':
+          src_btn = 'tsk';
+          break;
+        case 'project#':
+          src_btn = 'prj';
+          break;
+        case 'experiment#':
+          src_btn = 'exp';
+          break;
+        default:
+          src_btn = 'sam';
+          break;
+      }
+      $.getJSON(
+        dataMenuItemsUrl,
+        {query: query},
+        function(data){
+          if(data) {
+            _.each($('.atwho-header-res .title .btn'), function(el) {
+              $element = $(el);
+              if(data[$element.data('filter')].length === 0) {
+                $element.prop('disabled', true);
+                $element
+                  .removeClass('btn-primary')
+                  .addClass('btn-default');
+                $('[data-filter="' + src_btn +'"]')
+                  .removeClass('btn-default')
+                  .addClass('btn-primary');
+              } else {
+                $element.prop('disabled', false);
+                if($element.data('filter') == src_btn) {
+                  $element
+                    .removeClass('btn-default')
+                    .addClass('btn-primary');
+                } else {
+                  $element
+                    .removeClass('btn-primary')
+                    .addClass('btn-default');
+                }
+              }
+            });
+          }
+        });
+    }
+
+    // reset the dropdown
+    function reinitializeOnListHide() {
+      Util.showHideBinding();
+
+      _.map($('.atwho-view'), function(el) {
+        $(el).off('hide');
+        $(el).on('hide', function() {
+          $(field).atwho('destroy');
+          init();
+        });
+      });
+    }
+
+    // initialise dropdown dismiss button
+    function initDismissButton() {
```

Comment:
This functionality doesn't work on `@` mentions (my code block). Can you also apply it there?